File: backend/api/src/main/scala/com/github/ai/split/api/response/PutMemberResponse.scala
package com.github.ai.split.api.response

import com.github.ai.split.api.GroupDto
import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder, JsonDecoder, JsonEncoder}

case class PutMemberResponse(
  group: GroupDto
)

object PutMemberResponse {
  implicit val encoder: JsonEncoder[PutMemberResponse] = DeriveJsonEncoder.gen[PutMemberResponse]
  implicit val decoder: JsonDecoder[PutMemberResponse] = DeriveJsonDecoder.gen[PutMemberResponse]
}

File: backend/api/src/main/java/com/github/ai/split/api/response/PutMemberResponse.java
package com.github.ai.split.api.response;

import com.github.ai.split.api.GroupDto;
public class PutMemberResponse {

    public GroupDto group = new GroupDto();

    public PutMemberResponse() {}

    public PutMemberResponse(GroupDto group) {
        this.group = group != null ? group : new GroupDto();
    }
}